vpd-tool/src/vpd_tool.cpp

```diff
@@ -105,13 +115,183 @@ int VpdTool::dumpObject(const std::string& i_fruPath) const noexcept
     return l_rc;
 }
 
-nlohmann::json VpdTool::getFruProperties(const std::string& i_fruPath) const
+nlohmann::json VpdTool::getFruProperties(const std::string& i_objectPath) const
+{
+    nlohmann::json l_fruJson = nlohmann::json::object_t({});
+
+    // check if FRU is present in the system
+    const bool l_isFruPresent = isFruPresent(i_objectPath);
```

Comment:
If it is a one time use variable, get rid of it. Just check the return value in place.